Documentation/UPI/vSphere/vSphere.md

```diff
@@ -0,0 +1,196 @@
+```
+
+### Create FCOS VMs
+```
+for vm in \
+    okd4-master-1 okd4-master-2 okd4-master-3 \
+    okd4-worker-1 okd4-worker-2 okd4-worker-3 \
+    okd4-bootstrap; do
+    govc vm.clone -vm fedora-coreos-31.20200113.3.1-vmware.x86_64 \
+        -ds <datastore_name> -folder <vm_folder> -on=false \
+        -c=4 -m=8192 -net=<network_name> $vm
+
+    govc vm.disk.change -vm $vm -disk.label "Hard disk 1" -size 120G
+done
+```
+
+### Configure DNS, DHCP and LB
+The installation requires specific configuration of DNS and a load balancer. The requirements are listend in the official Openshift documentation: [Creating the user-provisioned infrastructure](https://docs.openshift.com/container-platform/4.2/installing/installing_vsphere/installing-vsphere.html#installation-infrastructure-user-infra_installing-vsphere). Example configurations are available at [requirements](/Documentation/UPI/Requirements)
+
+You will also need working DHCP on the network the cluster hosts are connected to. The DHCP server should assign the hosts unique FQDNs.
+
+### Create cluster configuration and Ignition files
+Create `install-config.yaml`:
+
+```
+apiVersion: v1
+baseDomain: domain.tld
+metadata:
+  name: cluster
+
+compute:
+- hyperthreading: Enabled
+  name: worker
+  replicas: 3
+
+controlPlane:
+  hyperthreading: Enabled
+  name: master
+  replicas: 3
+
+platform:
+  none: {}
+
+pullSecret: '<Your pull secret from https://cloud.redhat.com/openshift/install/vsphere/user-provisioned>'
+sshKey: <Your public SSH key beginning with ssh-rsa ...>
+```
+
+**NOTE**: It is a good idea to keep a copy of `install-config.yaml` for if you need to recreate the Ignition files since the following step destroys it.
+
+
+Generate Ignition-configs:
+
+```
+openshift-install create ignition-configs
+```
+
+Your `install-config.yaml` file should have been replaced with a bunch of `.ign` files which will be used to configure the FCOS hosts. 
+
+**NOTE**: If you want to totally regenerate the Ignition configs (for example to replace expired temporary certificates) you will also need to remove a hidden `.openshift_install_state.json`-file
+
+### Serve bootstrap.ign
+
+Due to the size of the bootstrap.ign file it can't be directly written into the VM metadata but needs to be served over HTTP instead. One way to do this is to use `python3 -m http.server`.
+
+Create a file `append-bootstrap.ign` which contains an URL to the full `bootstrap.ign`:
+
+```
+{
+  "ignition": {
+    "config": {
+      "merge": [
+        {
+          "source": "http://10.0.0.50:8000/bootstrap.ign", 
+          "verification": {}
+        }
+      ]
+    },
+    "timeouts": {},
+    "version": "3.0.0"
+  }
+}
+```
+
+### Set the VM metadata
+Steps which need to be done:
+- Set the VM property `guestinfo.ignition.config.data` to a base64-encoded version of the Ignition-config
+- Set the VM property `guestinfo.ignition.config.data.encoding` to `base64`
+- Set the VM property `disk.EnableUUID` to `TRUE`
```

Comment:
in our CI we don't enable this setting. What does it do?

Reply:
Do you mean `disk.EnableUUID`? 

From https://access.redhat.com/solutions/4606201:

> The parameter disk.EnableUUID is necessary so that the VMDK always presents a consistent UUID to the VM, thus allowing the disk to be mounted properly.

I don't know if it is necessary, I got it from the OCP docs at [Creating Red Hat Enterprise Linux CoreOS (RHCOS) machines in vSphere](https://docs.openshift.com/container-platform/4.2/installing/installing_vsphere/installing-vsphere.html#installation-vsphere-machines_installing-vsphere)

Reply:
Oh, okay, lets keep it then